### PR TITLE
Move do_scroll from Editor to View

### PR DESCRIPTION
**Pull request by cmyr**

### CODE REVIEW

File: rust/core-lib/src/view.rs [hunk 1]
```diff
@@ -283,7 +277,7 @@ impl View {
             );
             sel
         });
-        new_sel.and_then(|new_sel| self.set_selection(text, new_sel))
+        new_sel.map(|sel| self.set_selection(text, sel));
```
Comment (raphlinus) [inline rust/core-lib/src/view.rs]: Total style point, but I'd probably write this out as
```
if let Some(sel) = new_sel {
    self.set_selection(text, sel);
}
```

File: rust/core-lib/src/view.rs [hunk 2]
```diff
@@ -486,6 +480,11 @@ impl View {
         let height = self.line_of_offset(text, text.len()) + 1;
         let plan = RenderPlan::create(height, self.first_line, self.height);
         self.send_update_for_plan(text, tab_ctx, style_spans, &plan);
+        if let Some(new_scroll_pos) = self.scroll_to {
```
Comment (raphlinus) [inline rust/core-lib/src/view.rs]: It's tempting to write `self.scroll_to.take() and save a line, but this is probably easier to understand.

File: rust/core-lib/src/view.rs [hunk 3]
```diff
@@ -586,7 +585,7 @@ impl View {
     ///
     /// Return value is a location of a point that should be scrolled into view.
```
Comment (raphlinus) [inline rust/core-lib/src/view.rs]: Need to update comment, something like "Requests the point that should be scrolled into view."

File: rust/core-lib/src/view.rs [hunk 4]
```diff
@@ -774,22 +773,29 @@ impl View {
...
     /// determines whether the search should stop at the first found occurrence (does only apply
     /// to forward search, i.e. reverse = false). If `allow_same` is set to `true` the current
     /// selection is considered a valid next occurrence.
     pub fn select_next_occurrence(&mut self, text: &Rope, reverse: bool, wrapped: bool,
-                                  stop_on_found: bool, allow_same: bool) -> Option<usize>
+                                  stop_on_found: bool, allow_same: bool)
     {
         if self.search_string.is_none() {
-            return None;
+            return
```
Comment (raphlinus) [inline rust/core-lib/src/view.rs]: I personally would include the trailing `;`

**Comment by cmyr** [comment]
Addressed your comments.

I've been noticing just how much of the code in editor.rs is just plumbing to access the view; I'm going to look into clearing that up more tomorrow.

**Pull request merged**